caproto/ioc_examples/thermo_sim.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+import argparse
+from caproto.server import pvproperty, PVGroup, template_arg_parser, run
+import numpy as np
+import time
+from textwrap import dedent
+
+
+class Thermo(PVGroup):
+    """
+    Simulates (poorly) an oscillating temperature controller.
+
+    Follows :math:`T_{output} = T_{var} exp^{-(t - t_0)/K} sin(ω t) + T_{setpoint}`
+
+    The default prefix is `thermo:`
+
+    Readonly PVs
+    ------------
+
+    readback -> the current value
+
+    Control PVs
+    -----------
+
+    setpoint -> where to go
+    K -> the decay constant
+    omega -> the oscillation frequency
+    Tvar -> the scale of the oscillations
+    """
+
+    def __init__(self, *args, period, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._T0 = time.monotonic()
+        self.period = period
+
+    readback = pvproperty(value=[0], dtype=float, read_only=True)
+
+    setpoint = pvproperty(value=[100], dtype=float)
+    K = pvproperty(value=[10], dtype=float)
+    omega = pvproperty(value=[(np.pi)], dtype=float)
+    Tvar = pvproperty(value=[10], dtype=float)
+
+    @readback.startup
+    async def readback(self, instance, async_lib):
+
+        def t_rbv(setpoint, K, omega, Tvar):
+            t = time.monotonic()
+            return ((Tvar *
+                     np.exp(-(t - self._T0) / K) *
+                     np.sin(omega * t)) +
+                    setpoint)
+
+        while True:
+            T = t_rbv(**{k: getattr(self, k).value[0]
+                         for k in ['setpoint', 'K', 'omega', 'Tvar']})
+
+            await instance.write(value=[T])
+            await async_lib.library.sleep(self.period)
+
+    @setpoint.putter
+    async def setpoint(self, instance, value):
+        self._T0 = time.monotonic()
+        return value
+
+
+if __name__ == '__main__':
+    parser, split_args = template_arg_parser(
+        default_prefix='thermo:',
+        desc=dedent(Thermo.__doc__))
```

Comment:
Agreed with @danielballan - though I get the desire to change docs in one place and one place only, this seems a bit excessive...